lightning/src/ln/channel.rs

```diff
@@ -1608,6 +1609,61 @@ pub(super) trait InteractivelyFunded<SP: Deref> where SP::Target: SignerProvider
 			}),
 		})
 	}
+
+	fn internal_funding_tx_constructed<L: Deref>(
+		&mut self, signing_session: &mut InteractiveTxSigningSession, logger: &L
+	) -> Result<(msgs::CommitmentSigned, Option<Event>), ChannelError>
+	where
+		L::Target: Logger
+	{
+		let our_funding_satoshis = self.dual_funding_context().our_funding_satoshis;
+
+		let mut output_index = None;
+		let expected_spk = self.context().get_funding_redeemscript().to_p2wsh();
+		for (idx, outp) in signing_session.unsigned_tx.outputs().enumerate() {
+			if outp.script_pubkey() == &expected_spk && outp.value() == self.context().get_value_satoshis() {
+				if output_index.is_some() {
+					return Err(ChannelError::Close(
+						(
+							"Multiple outputs matched the expected script and value".to_owned(),
+							ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+						)));
+				}
+				output_index = Some(idx as u16);
+			}
+		}
+		let outpoint = if let Some(output_index) = output_index {
+			OutPoint { txid: signing_session.unsigned_tx.txid(), index: output_index }
+		} else {
+			return Err(ChannelError::Close(
+				(
+					"No output matched the funding script_pubkey".to_owned(),
+					ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+				)));
+		};
+		self.context_mut().channel_transaction_parameters.funding_outpoint = Some(outpoint);
+		let channel_transaction_parameters = self.context().channel_transaction_parameters.clone();
+		self.context_mut().holder_signer.as_mut().provide_channel_parameters(&channel_transaction_parameters);
+
+		let commitment_signed = get_initial_commitment_signed(self.context_mut(), signing_session.unsigned_tx.clone(), logger);
```

Comment:
`get_initial_commitment_signed` can take the transaction by reference to avoid this clone.

Reply:
Yeah, lol